### ISSUE TITLE
Create a way to control log message output.

### ISSUE THREAD

**Issue by rekhajoshm** [tensorflow/tensorflow#1258]
Create a way to control log message output., most options should be passed via ConfigProto or another proto. Maybe add a LogOptions structure to hold options related to logging.  #1229
### Environment info

Operating System: Mac 10.9.5
### Feature request
1. Have a way to control logs that you record.As in gflags/re2 flags.Refer  #1229 
### What have you tried?
1. #1229

**Comment by vrv** [comment]
Okay, see https://github.com/tensorflow/tensorflow/commit/0ed66eb560957987901eef40dcf8e74247841528 for how you can set a min log level for logging in C++ using an environment variable.

Defaults to 0, so all logs are shown.  Set TF_CPP_MIN_LOG_LEVEL to 1 to filter out INFO logs, 2 to additionall filter out WARNING, 3 to additionally filter out ERROR.

**Comment by rbunn80110** [comment]
What about setting a min logging level in Python?

**Comment by yaroslavvb** [comment]
You can do it using `os.environ` 

```
import os
os.environ['TF_CPP_MIN_LOG_LEVEL']='3'
import tensorflow as tf

```

**Comment by FeralWhippet** [comment]
just got 12.1 of tensorflow-gpu for python on windows 10 (64 bit), installed via pip, and setting TF_CPP_MIN_LOG_LEVEL has no effect on the log messages. I have tried via environment variable in the shell and via setting it in the python script as above. Is this a known issue for windows?

**Comment by yaroslavvb** [comment]
It's not in 12.1, you need a later version (nightly)

**Comment by abrahamrhoffman** [comment]
@yaroslavvb : and there was MUCH rejoicing! 👯 👏

**Comment by russellanam** [comment]
Hi, I am using tf 12.1. I compiled it from the master branch. I have set the following in my python code: 

`import os
os.environ['TF_CPP_MIN_LOG_LEVEL']='0'`

but I don't get any VLOG outputs from the C code.

Specifically I am after this particular log (tensorflow/core/platform/default/conv_op.cc)

VLOG(2) << "Conv2D: in_depth = " << in_depth
            << ", input_cols = " << input_cols
            << ", filter_cols = " << filter_cols
            << ", input_rows = " << input_rows
            << ", filter_rows = " << filter_rows
            << ", stride_rows = " << stride_rows
            << ", stride_cols = " << stride_cols
            << ", out_depth = " << out_depth;

I do get INFO logs from python part like this:

INFO:tensorflow:Summary name loss/cross_entropy (raw) is illegal; using loss/cross_entropy__raw_ instead.

But nothing from the compiled C library.

**Comment by yaroslavvb** [comment]
It's TF_CPP_MIN_VLOG_LEVEL, not TF_CPP_MIN_LOG_LEVEL
Also, note that if TF_CPP_MIN_LOG_LEVEL is set, then TF_CPP_MIN_VLOG_LEVEL
values are ignored

**Comment by yaroslavvb** [comment]
Also, not sure if 12.1 is late enough, you may need 1.0rc1 or later

**Comment by russellanam** [comment]
Thanks for the quick reply. I can confirm that TF_CPP_MIN_VLOG_LEVEL does not work with v0.12. I will try with 1.0rc1.

**Comment by dbuendiab** [comment]
It works with TF v0.12.1 
![image](https://cloud.githubusercontent.com/assets/3029862/23041760/52a7a074-f496-11e6-907c-d36a0f73e812.png)

**Comment by gongzhitaao** [comment]
I can confirm the following works with TF v1.0.0
```python
import os
os.environ['TF_CPP_MIN_LOG_LEVEL']='3'
import tensorflow as tf
```

**Comment by deepdad** [comment]
WARNING:tensorflow:From /usr/local/lib/python3.5/dist-packages/tensorflow/contrib/learn/python/learn/estimators/dnn.py:374: calling BaseEstimator.predict (from tensorflow.contrib.learn.python.learn.estimators.estimator) with x is deprecated and will be removed after 2016-12-01

#tf.logging.set_verbosity(tf.logging.FATAL)#INFO
import os
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '5'#0

can't suppress from tflearn? they seem more like printed strings

**Comment by AndreyPlotkinOr** [comment]
For one interested, in C++ you may manage TF logging with this line of code:
_putenv_s("TF_CPP_MIN_LOG_LEVEL", "1");
Log levels are defined in tensorflow\core\platform\default\logging.h

**Comment by fperezgamonal** [comment]
Hi @AndreyPlotkinOr . I've tried your approach and I cannot hide the:
 `2020-07-03 13:31:19.012804: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll` type of messages with any log level (1, 2 or 3)

Do you mind sharing where did you call the function? Or it should not matter? In my case I call it very early in the `main` function.
Mind you it may be that these types of messages cannot be hidden by means of the logging flags.

Cheers!

PS: I'm using 1.15.3 build from source on Windows if it helps! :)

Ferran.
